### DISCUSSION TITLE
Should I use Janus server or embed it as a library?

### DISCUSSION THREAD

**Discussion by melgenek** [JanusGraph/janusgraph#2498] [Q&A] [answered]
Hi, Janus community!

I am going to deploy Janus with Scylla backend.
The graph is going to be used by:
1) an http app that does simple queries by node ids
2) long-running traversals in spark

It seems that there are two options to run Janus:
1) as a cluster of docker containers 
2) as java library

Approach 1 works better for me because it allows hosting Janus as a regular database where the clients simply send requests via a Gremlin api.

There are some restrictions imposed by the remote gremlin connection as described here https://tinkerpop.apache.org/docs/current/reference/#connecting-gremlin-server-limitations. Though, it's not really clear in which use cases Janus server or embedded mode should be used.

Is it a viable option to host a set docker containers for my use case or should I consider using Janus as a library?

Thank you in advance for your help!

**Comment by li-boxuan** [comment, marked as answer]
FYI, there is a related discussion: https://github.com/JanusGraph/janusgraph/issues/1108

**Reply by melgenek** [reply]
Thank you a lot. It's a great resource that describes pros and cons of both approaches.

As a summary for myself and others who come to this issue later, I'll post the @FlorianHockmann's recommendation from this post https://groups.google.com/g/janusgraph-users/c/t7gNBeWC844/m/gW7QzowECAAJ:
> The overall recommendation is to use JanusGraph Server, unless you have really good reasons to use JanusGraph embedded.
